src/main/scala/User.scala

```diff
@@ -1,3 +1,120 @@
-object User {
+import DataBaseConn.{getAll, putUser}
+import User.registerAndLoginRoute
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import spray.json._
 
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+/*
+  POST /register -> register User(username, password)
+  GET /users -> get all the users
+  GET /user/{id} -> get a user by id
+  GET /login -> OK (Successful), NotFound (Error)
+ */
+
+// id is auto incremented
+case class UserComplete(id: Int, username: String, password: String)
+case class UserPartial(username: String, password: String)
+
+trait UserJsonProtocol extends DefaultJsonProtocol {
+  implicit val userPartialFormat = jsonFormat2(UserPartial)
+  implicit val userCompleteFormat = jsonFormat3(UserComplete)
+}
+
+trait ActorUser{
+  implicit val system = ActorSystem("UserActor")
+}
+
+object User extends UserJsonProtocol with ActorUser {
+
+  import system.dispatcher
+
+  def checkLoginUser(username: String, password: String): Boolean = {
+    if(!(getAll.find(i => i.username == username && i.password == password).isEmpty)) true
+    else false
+  }
+
+  def findUserById(idUser: Int) = {
+    getAll.find(i => i.id == idUser)
+  }
+
+  def registerHashPass(password: String): String = password.hashCode.toString
+
+  val registerAndLoginRoute =
+    (path("register") & post & extractRequest & extractLog){ (request, log) =>
+      //TODO: try to use .map/for
+      val entity = request.entity
+      val strictEntity = entity.toStrict(2 seconds)
+      val userFuture = strictEntity.map(_.data.utf8String.parseJson.convertTo[UserPartial])
+
+      if(getAll.find(i => i.username == userFuture.value.get.get.username).isEmpty) {
+        userFuture.onComplete {
+          case Success(value) =>
+            putUser(value.username, registerHashPass(value.password))
+          case Failure(exception) =>
+            log.error(s"Error: $exception")
+        }
+        complete(userFuture
+          .map(_ => StatusCodes.OK)
+          .recover {
+            case _ => StatusCodes.InternalServerError
+          }
+        )
+      }
+      else complete(StatusCodes.Forbidden)
+    }~
+      (path("users") & get){
+        complete(
+          HttpEntity(
+            ContentTypes.`application/json`,
+            getAll.toJson.prettyPrint
+          )
+        )
+      }~
+      (path("user" / IntNumber ) & get){ userId =>
+        val maybeId: Option[UserComplete] = findUserById(userId)
+        maybeId match {
+          case Some(item) =>
+            complete(
```

Comment:
Why would we return the password on the front end? I think we can return only a code (Ok, maybe Accepted - with the _meaning_ that the credentials were corrected).